### ISSUE TITLE
pre-commit post-rewrite failure ends with detatched head

### ISSUE THREAD

**Issue by shareefj** [pre-commit/pre-commit#2397]
### search tried in the issue tracker

yes

### describe your issue

I have a hook designed to catch changes to a requirements.txt file and alert the user when they run a clone, merge or rebase.  However, when doing a rebase, mostly I've been doing `git pull --rebase`, the hook fires and stops the rebase from finishing and I'm left in a detached head state.

The only way I've so far been able to work around this is by deleting all `.git/hooks/*` and running again.

Is there a trick I'm missing to get a rebase to complete?

### pre-commit --version

pre-commit 2.17.0

### .pre-commit-config.yaml

```yaml
default_stages: [commit, push]
exclude: 'vendor/'
repos:
- repo: local
  hooks:
  - id: watch-requirements
    name: Watches requirements.txt for changes
    language: script
    entry: ./hooks/alert_if_modified.py
    always_run: true
    args:
    - --filename
    - requirements.txt
    - --message
    - |
       Alert : requirements.txt has been modified; update your venv.
               try running `create_venv --rebuild $VIRTUAL_ENV`
    stages: [post-checkout, post-merge, post-rewrite]
```

**Comment by asottile** [comment]
I don't use `post-rewrite` myself but usually when a rebase is interrupted you'll resolve whatever is broken and `git rebase --continue`

**Comment by shareefj** [comment]
Yes, well, if it got to that point then I'd be happy.  It looks like the pre-commit hook causes Git to barf in the middle of rebasing.  I just did a search and found a similar issue here: https://stackoverflow.com/questions/25561485/git-rebase-i-with-squash-cannot-detach-head

I tried running with `--no-verify` but that doesn't seem to be obeyed for a rebase.  I'm not a Git expert but is that expected?
